source/common/formatter/substitution_formatter.cc

```diff
@@ -1918,6 +1925,69 @@ UpstreamHostMetadataFormatter::UpstreamHostMetadataFormatter(const std::string&
                           return host->metadata().get();
                         }) {}
 
+ProxyProtocolTlvsFormatter::ProxyProtocolTlvsFormatter(const std::string& tlv_type) {
+  ASSERT(!tlv_type.empty());
+  // Specified tlv_type must be parsable as an int.
+  try {
+    tlv_type_ = std::stoi(tlv_type);
+  } catch (const EnvoyException& ex) {
```

Comment:
🤦‍♀️ when I wrote this I thought that this was catching cases where the provided param is invalid / not a number. I saw many uses of catching `EnvoyException` throughout the repo, so I thought it might be preferable to catch this wrapper exception type instead of the more specific `[std::out_of_range](https://en.cppreference.com/w/cpp/error/out_of_range)` or [std::invalid_argument](https://en.cppreference.com/w/cpp/error/invalid_argument) exceptions. I see now that it doesn't wrap around `std` library exceptions. I must've not re-built the binary before manually testing this part (this was the last change I made). Whoops! 

I'll revert the exception back to `const std::invalid_argument&`.